expr/expr.go

```diff
@@ -13,15 +13,17 @@ const (
 	etName   exprType = iota // a string without quotes, e.g. metric.name, metric.*.query.patt* or special values like None which some functions expect
 	etBool                   // True or False
 	etFunc                   // a function call like movingAverage(foo, bar)
-	etConst                  // any number, parsed as a float64 value
+	etInt                    // any number with no decimal numbers, parsed as a float64 value
+	etFloat                  // any number with decimals, parsed as a float64 value
 	etString                 // anything that was between '' or ""
 )
 
 // expr represents a parsed expression
 type expr struct {
 	etype     exprType
-	float     float64          // for etConst
-	str       string           // for etName, etFunc (func name), etString, etBool and etConst (unparsed input value)
+	float     float64          // for etFloat
+	i         int64            // for etInt
```

Comment:
i didn't do this initially cause they are golang keywords, but i tested it now and it works fine, it's just that the editor highlights them as keywords but other than that it's harmless and I agree, better.